subsys/usb/device_next/usbd_endpoint.c

```diff
@@ -125,7 +125,12 @@ int usbd_ep_enqueue(const struct usbd_class_data *const c_data,
 
 	if (USB_EP_DIR_IS_IN(bi->ep)) {
 		if (usbd_is_suspended(uds_ctx)) {
-			return -EPERM;
+			int ret;
+
+			ret = usbd_wakeup_request(uds_ctx);
+			if (ret != 0) {
+				return ret;
```

Comment:
Should we really be ending enqueue with an error if device is suspended (and wakeup fails or feature is cleared by the host)?

Reply:
Good question, if it is okay to enqueue in suspended state, then should we call remote wakeup signaling here or from the driver?

Reply:
Linux dwc2 gadget driver seems to allow queuing transfers only in active state (L0). If the device is in any other state (suspended), then it fails the queue with `-EAGAIN`. But it really depends on the gadget driver - some do fail when suspended, and some do not care.